src/foam/u2/crunch/CapabilityCardView.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.u2.crunch',
+  name: 'CapabilityCardView',
+  extends: 'foam.u2.View',
+
+  implements: [ 'foam.mlang.Expressions' ],
+
+  requires: [
+    'foam.u2.Element',
+    'foam.u2.crunch.Style',
+    'foam.nanos.crunch.UserCapabilityJunction',
+    'foam.nanos.crunch.CapabilityJunctionStatus',
+    'foam.u2.view.ReadOnlyEnumView',
+  ],
+
+  imports: [
+    'user',
+    'userCapabilityJunctionDAO',
+  ],
+
+  documentation: `
+      A single card in a list of capabilities.
+  `,
+
+  methods: [
+    function initE() {
+      this.SUPER();
+      var self = this;
+
+      // Methods of Style all return the first argument for chaining
+      self.s = self.Style.create();
+      
+      self
+        .s.addClassTo(self)
+        .s.addClassTo(self, 'mode-circle')
+        .start()
+          .addClass(self.s.myClass('icon-circle'))
+          .style({
+            'background-image': "url('" + self.data.icon + "')",
+            'background-size': 'cover',
+            'background-position': '50% 50%',
+          })
+        .end()
+        .start('span')
+        .call(function () {
+          var badgeWrapper = self.Element.create({ nodeName: 'SPAN' });
+          this.add(badgeWrapper);
+          self.userCapabilityJunctionDAO.find(self.AND(
+            self.EQ(self.UserCapabilityJunction.SOURCE_ID, self.user.id),
+            self.EQ(self.UserCapabilityJunction.TARGET_ID, self.data.id),
+          )).then(ucj => {
+            if ( ! ucj ) {
+              var badge = self.Element.create();
+              badge
+                .addClass(self.s.myClass('badge'))
+                .addClass(self.s.myClass('badge-info'))
+                .add("available")
+                ;
+              badgeWrapper.add(badge);
+            }
+            else {
+              var badge = self.ReadOnlyEnumView.create({
+                data: ucj.status
+              })
+                .addClass(self.s.myClass('badge'))
+                .style({ 'background-color': ucj.status.background })
+                ;
```

Comment:
I like hanging the semicolon here like this for two reasons:
- Easy to move lines without worrying about ensuring the last one has the semicolon
- Adding a line to the end doesn't affect the previous line in version control